### PR TITLE
landing_worker: move merge conflict processing to function (bug 1823898)

### PR DESCRIPTION
**Pull request by zzzeid**
Move merge conflict processing to its own function.

### CODE REVIEW

File: landoapi/workers/landing_worker.py [hunk 1]
```diff
@@ -148,6 +148,56 @@ def notify_user_of_landing_failure(job):
...
+        breakdown["failed_paths"] = [
+            {
+                "path": r[0],
+                "url": f"{repo.pull_path}/file/{r[1].decode('utf-8')}/{r[0]}",
+                "changeset_id": r[1].decode("utf-8"),
+            }
+            for r in failed_path_changesets
+        ]
+        breakdown["reject_paths"] = {}
+        for r in reject_paths:
```
Comment (cgsheeh) [inline landoapi/workers/landing_worker.py:189]: r -> `path`, or `reject_path`

**Review by cgsheeh — CHANGES_REQUESTED** [review]
How difficult would it be to add a test for this function? If possible we should add one now.

File: landoapi/workers/landing_worker.py [hunk 2]
```diff
@@ -148,6 +148,56 @@ def notify_user_of_landing_failure(job):
             job.requester_email, job.head_revision, job.error, job.id
         )
 
+    def process_merge_conflict(self, exception, repo, hgrepo, revision_id):
```
[the remark this answers is not in the document]
Comment (zzzeid) [reply landoapi/workers/landing_worker.py:151]: I think it makes sense to add `flake8-docstrings` to help with these checks. I've already added it to other repos, and it makes sense to enforce this as a standard. Will file a bug.

**Comment by zzzeid** [comment]
> > How difficult would it be to add a test for this function? If possible we should add one now.
> 
> There are already integration tests that test some of this code (e.g., `test_landings.py::test_landing_worker__extract_error_data`). We could add a unit test for this in the future but I think it's out of scope for this PR (as this is mainly just extracted from #224 and mainly just moves the exact code block to its own function.)

Also I am planning some improvements to this code as it will ultimately be reused by the revision worker to detect conflicts. That might be a better time to write unit tests for this.

File: landoapi/workers/landing_worker.py [hunk 3]
```diff
@@ -148,6 +148,63 @@ def notify_user_of_landing_failure(job):
...
         )
 
+    def process_merge_conflict(
+            self,
+            exception: PatchConflict,
+            repo: Repo,
+            hgrepo: HgRepo,
+            revision_id: int,
+    ):
+        "Extract and parser merge conflict data from exception into a usable format."""
```
Comment (zzzeid) [inline landoapi/workers/landing_worker.py:158]:
```suggestion
        "Extract and parse merge conflict data from exception into a usable format."""
```

File: landoapi/workers/landing_worker.py [hunk 4]
```diff
@@ -148,6 +148,63 @@ def notify_user_of_landing_failure(job):
             job.requester_email, job.head_revision, job.error, job.id
         )
 
+    def process_merge_conflict(
+        self,
+        exception: PatchConflict,
+        repo: Repo,
+        hgrepo: HgRepo,
+        revision_id: int,
+    ):
```
Comment (cgsheeh) [inline landoapi/workers/landing_worker.py:157]:
```suggestion
    def process_merge_conflict(
        self,
        exception: PatchConflict,
        repo: Repo,
        hgrepo: HgRepo,
        revision_id: int,
    ) -> dict[str, Any]:
```

**Review by cgsheeh — CHANGES_REQUESTED** [review]

File: landoapi/workers/landing_worker.py [hunk 5]
```diff
@@ -148,6 +149,63 @@ def notify_user_of_landing_failure(job):
...
         )
 
+    def process_merge_conflict(
+        self,
+        exception: PatchConflict,
+        repo: Repo,
+        hgrepo: HgRepo,
+        revision_id: int,
+    ) -> dict[str, Any]:
+        "Extract and parse merge conflict data from exception into a usable format." ""
```
Comment (cgsheeh) [inline landoapi/workers/landing_worker.py:159]:
```suggestion
        """Extract and parse merge conflict data from exception into a usable format."""
```

**Review by cgsheeh — CHANGES_REQUESTED** [review]

File: landoapi/workers/landing_worker.py (hunk 5 above)
Comment (zzzeid) [reply landoapi/workers/landing_worker.py]: Nice catch, interesting how this slipped through the linters, most likely black automatically added the space at the end.

**Pull request merged**